pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -53,13 +54,29 @@ func chunkSizeFor(n, parallelism int) int {
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece func(piece int) error, operation string) error {
 	goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
+	goroutinesDurationMetric := metrics.GoroutinesDuration.WithLabelValues(operation)
+	errCh := NewErrorChannel()
+	ctx, cancel := context.WithCancel(ctx)
+
 	withMetrics := func(piece int) {
+		startTime := time.Now()
+		result := "success"
 		goroutinesMetric.Inc()
-		doWorkPiece(piece)
+		if err := doWorkPiece(piece); err != nil {
+			result = "error"
+			errCh.SendErrorWithCancel(err, cancel)
+		}
+		metrics.GoroutinesExecutionTotal.WithLabelValues(operation, result).Inc()
```

Comment:
I think need to move this out and make it similar to 
```
goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
```

Context: https://github.com/kubernetes/kubernetes/issues/116427

Reply:
Right, @Gekko0114 can you check `.WithLabelValues(...).WithLabelValues(...)` works intentionally? If Yes, we can do ↑.